File: exercice6/main.ts
// Exercice 6 : Créez une classe ClientProfessionnel qui étend la classe Client et qui a comme attribut un numéro SIRET. 

class Client {

    protected fullName: string;
    protected accountID: number;

    constructor(fullName: string, accountID: number) {
        this.fullName = fullName;
        this.accountID = accountID;
    }

    getFullName(): string {
        return this.fullName;
    }

    getAccountID(): number {
        return this.accountID;
    }

};

class ProfessionalClient extends Client {

    siretNumber: number;

    constructor(fullName: string, accountID: number, siretNumber: number) {
        super(fullName, accountID);
        this.siretNumber = siretNumber;
    }

    getSiretNumber(): number {
        return this.siretNumber;
    }

}

let firstClient: Client = new Client("Bill Gates", 537);
let secondClient: ProfessionalClient = new ProfessionalClient("Jeff Bezos", 777, 39233923923);

console.log("Nom du client : " + firstClient.getFullName());
console.log("N° de compte : " + firstClient.getAccountID());

console.log("Nom du client : " + secondClient.getFullName());
console.log("N° de compte : " + secondClient.getAccountID());
console.log("N° de siret : " + secondClient.getSiretNumber());

File: exercice6/main.java
class Client {
    
    String fullName;
    int accountID;
    
    public Client(String fullName, int accountID) {
        
        this.fullName = fullName;
        this.accountID = accountID;
    }

}

class ProfessionalClient extends Client {
    int siretNumber;
    
    ProfessionalClient(String fullName, int accountID, int siretNumber) {
            super(fullName, accountID);
            this.siretNumber = siretNumber;
    }
}


class Playground {
    public static void main(String[ ] args) {
        Client firstClient = new Client ("Bill Gates", 628);
        ProfessionalClient secondClient = new ProfessionalClient ("Jeff Bezos", 777, 23823838);
        System.out.println("Nom du client : " + firstClient.fullName);
        System.out.println("N° de compte : " + firstClient.accountID);
        
        System.out.println("Nom du client : " + secondClient.fullName);
        System.out.println("N° de compte : " + secondClient.accountID);
        System.out.println("N° de siret : " + secondClient.siretNumber);
    }
}